cymetric/convenient_interface.py

```diff
@@ -0,0 +1,650 @@
+#!/usr/bin/env python
+
+import warnings
+
+import pandas as pd
+import numpy as np
+
+import cymetric as cym
+from cymetric import tools
+
+try:
+    from pyne import data
+    import pyne.enrichment as enr
+    from pyne import nucname
+    HAVE_PYNE = True
+except ImportError:
+    HAVE_PYNE = False
+
+try:
+    from graphviz import Digraph
+    HAVE_GRAPHVIZ = True
+except ImportError:
+    HAVE_GRAPHVIZ = False
+
+
+def format_nuclist(nuc_list):
+    """
+    format the nuclide list provided by the users into a standart format:
+    ZZAAIIIII.
+    Parameters
+    ----------
+    nuc_list:  list of nuclides
+    """
+    tools.raise_no_pyne('Unable to format nuclide list!', HAVE_PYNE)
+
+    for i in range(len(nuc_list)):
+        nuc_list[i] = nucname.id(nuc_list[i])
+
+    return nuc_list
+
+
+def add_missing_time_step(df, ref_time):
+    """
+    Add the missing time step to a Panda Data Frame.
+    Parameters
+    ----------
+    df:  Pandas Data Frame
+    ref_time:  list of the time step references (Coming from TimeStep metrics)
+    """
+    ref_time.rename(index=str, columns={'TimeStep':  'Time'}, inplace=True)
+
+    if 'SimId' in ref_time.columns.values:
+        ref_time.drop('SimId', 1, inplace=True)
+    df = pd.merge(ref_time, df, how="outer")
+    df.fillna(0, inplace=True)
+    return df
+
+
+def merge(df, base_col, add_df, add_col):
+    """
+    Merge some additionnal columns fram an additionnal Pandas Data Frame
+    onother one and then remove the second base column (keeping SimID
+    information).
+    Parameters
+    ----------
+    df: Pandas Data Frame
+    base_col: list of the base columns names
+    add_df: Pandas Data Frame to add in the df one
+    add_col: columns to be added
+    """
+    df = pd.merge(add_df[add_col], df, on=base_col)
+    df.drop(base_col[1], 1)
+    return df
+
+
+def reduce(df, rdc_list):
+    """
+    Filter the df Pandas Data Frame according to the rdc_list (list of item
+    in the corresponding columns).
+    Parameters
+    ----------
+    df: Pandas Data Frame
+    rdc_list: list of pair of string and string list.
+    """
+    for rdc in rdc_list:
+        if len(rdc[1]) != 0:
+            df = df[df[rdc[0]].isin(rdc[1])]
+        else:
+            wng_msg = "Empty list provided for " + rdc[0] + " key."
+            warnings.warn(wng_msg, UserWarning)
+    return df
+
+
+def get_transaction_df(evaler_, send_list=[], rec_list=[], commod_list=[]):
```

Comment:
Also, mutable data types should almost never be the default values of keyword arguments. Please make these empty tuples `()` or `None`.